api/api.proto

```diff
@@ -2517,18 +2611,69 @@ message CloudMigrateStartRequest {
     string task_id = 4;
 }
 
+// Defines a migration request 
+message SdkCloudMigrateStartRequest {
+  // Defines a migration request for a volume
+  message MigrateVolume { 
+    string volume_id = 1;
+  }
+
+  // Defines a migration request for a volume group
+  message MigrateVolumeGroup { 
+    string group_id = 1;
+  }
+  
+  // Defines a migration request for all volumes in a cluster
+  message MigrateAllVolumes { 
+  }
+
+  // ID of the cluster to which volumes are to be migrated
+  string cluster_id = 1;
+  
+  // Unique name assocaiated with this migration. 
+  // This is a required field for idempotency
+  string name = 2;
```

Comment:
This is not a name. It is a task ID and that is what is returned when enumerating migrations.

Reply:
Can two migrations be created with the same task id? My guess is that they cannot, please let me know if I am wrong. The SDK has all idempotency values called as `name`. So this function would have to adjust to that. We will internally change it to task_id.

Reply:
The MigrateInfo returns taskIds, while this would take in a name. So this will cause confusion.

Reply:
It will not cause confusion at all because Volume.Create takes `name` and it provides a `volume_id`. This is the same.

Reply:
It is not the same. For volumes there are 2 separate fields, name and Id. For migrations there is just one field i.e taskId, there is no name returned in the migrateInfo.

Reply:
Talked offline, and we will rename this back to task_id and make it optional. This will allow the caller to decide if they want it idempotent or not. The documentation will show this and that it is optional.

Also, we need an issue for the cloudbackup change `name` to `task_id`.